lib-extra/src/test/java/com/diffplug/spotless/extra/wtp/EclipseWtpFormatterCommonTests.java

```diff
@@ -24,80 +25,45 @@
 import java.util.Arrays;
 import java.util.Properties;
 import java.util.function.Consumer;
+import java.util.stream.Stream;
 
-import org.junit.Test;
-import org.junit.runner.RunWith;
-import org.junit.runners.Parameterized;
-import org.junit.runners.Parameterized.Parameter;
-import org.junit.runners.Parameterized.Parameters;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.EnabledForJreRange;
 
 import com.diffplug.spotless.FormatterStep;
-import com.diffplug.spotless.JreVersion;
-import com.diffplug.spotless.TestProvisioner;
 import com.diffplug.spotless.extra.EclipseBasedStepBuilder;
 import com.diffplug.spotless.extra.eclipse.EclipseCommonTests;
 
-@RunWith(value = Parameterized.class)
-public class EclipseWtpFormatterStepTest extends EclipseCommonTests {
-
-	private enum WTP {
-		// @formatter:off
-		CSS(	"body {\na: v;   b:   \nv;\n}  \n",
-				"body {\n\ta: v;\n\tb: v;\n}"),
-		HTML(	"<!DOCTYPE html> <html>\t<head> <meta   charset=\"UTF-8\"></head>\n</html>  ",
-				"<!DOCTYPE html>\n<html>\n<head>\n<meta charset=\"UTF-8\">\n</head>\n</html>\n"),
-		JS(		"function f(  )   {\na.b(1,\n2);}",
-				"function f() {\n    a.b(1, 2);\n}"),
-		JSON(	"{\"a\": \"b\",	\"c\":   { \"d\": \"e\",\"f\": \"g\"}}",
-				"{\n\t\"a\": \"b\",\n\t\"c\": {\n\t\t\"d\": \"e\",\n\t\t\"f\": \"g\"\n\t}\n}"),
-		XML(	"<a><b>   c</b></a>", "<a>\n\t<b> c</b>\n</a>");
-		// @formatter:on
-
-		public final String input;
-		public final String expectation;
-
-		private WTP(String input, final String expectation) {
-			this.input = input;
-			this.expectation = expectation;
-		}
-
-		public EclipseBasedStepBuilder createBuilder() {
-			EclipseWtpFormatterStep stepType = EclipseWtpFormatterStep.valueOf(this.toString());
-			return stepType.createBuilder(TestProvisioner.mavenCentral());
-		}
-	}
+@EnabledForJreRange(min = JAVA_11)
+abstract class EclipseWtpFormatterCommonTests extends EclipseCommonTests {
```

Comment:
I did not find a good way to combine parametrization with test inheritance in JUnit 5. That's why I split this test class into:
 * `EclipseWtpFormatterCssTest`
 * `EclipseWtpFormatterHtmlTest`
 * `EclipseWtpFormatterJsonTest`
 * `EclipseWtpFormatterJsTest`
 * `EclipseWtpFormatterXmlTest`

separate test classes. Each class defines language-specific input and test expectation. Separate test classes replace the `WTP` enum.

I also effectively removed `EclipseWtpFormatterStepTestOld` because of the same reason. Method `#getSupportedVersions()` now returns both old and new versions in a single array.

Does this seem like a sensible approach? Suggestions are much appreciated

Reply:
I think the old stuff looks better, but the new stuff isn't bad. Adopting JUnit 5 is quite nice, and I think it's worth merging this quickly to avoid conflicts. Can always refactor back together in the future. This okay with you @fvgh?

Reply:
I dug into this a bit more and made a small change in f35a109. I think we're now testing everything that we tested before, and we should be good to go, but I'd still like @fvgh to sign-off first. Wanna press the merge button @fvgh when/if you're okay with this test change?

Reply:
LGTM. A little overhead having so many test classes, but it is readable.
@lutovich Thanks for putting in so much effort. I always missed JUnit5 functionality.